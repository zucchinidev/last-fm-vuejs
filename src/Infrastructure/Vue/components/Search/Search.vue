<template>
  <v-text-field
    :label="label"
    :value="model"
    single-line
    full-width
    @input="setValueOfInput"
    @keyup.enter="search"
    color="teal"
    prepend-icon="search"
  ></v-text-field>
</template>
<script>
export default {
  name: 'Search',
  data () {
    return {
      model: this.value
    }
  },
  props: {
    label: String,
    trim: {
      type: Boolean,
      default: true
    },
    value: String
  },
  methods: {
    setValueOfInput (event) {
      let value = event && event.target ? event.target.value : event
      if (this.trim && value) {
        value = value.trim()
      }
      this.model = value
      if (this.model === '') {
        this.$emit('input', this.model)
      }
    },
    search () {
      this.$emit('input', this.model)
    }
  }
}
</script>
<style scoped lang="stylus">
  li
    display block
    margin 10px 0
</style>
