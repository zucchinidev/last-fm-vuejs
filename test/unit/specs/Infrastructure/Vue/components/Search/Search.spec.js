import { WrapComponent } from '../../../../../helpers'
import Search from '@Vue/components/Search/Search'
import { SearchPageObject } from './SearchPageObject'

describe('Search.vue', () => {
  let wrapper
  let value
  let page
  let props
  beforeEach(() => {
    const spy = jest.spyOn(console, 'error').mockImplementation(() => {})
    value = '       fake_value'
    props = { value }
    wrapper = WrapComponent(Search).withProps(props).mount()
    page = new SearchPageObject(wrapper)
    spy.mockReset()
  })

  it('should correctly insert the value in the input field', () => {
    expect(page.getInputValue()).toEqual(value)
    page.fireInputEvent()
    page.fireSearch()
    expect(page.getTrimInputValue()).toEqual(value.trim())
    expect(page.getEmittedEvent('input')).toBeTruthy()
  })

  it('has the expected html structure', () => {
    page.matchSnapshot()
  })
})
