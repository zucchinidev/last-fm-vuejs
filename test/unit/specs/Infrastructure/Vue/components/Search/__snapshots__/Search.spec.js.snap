// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search.vue has the expected html structure 1`] = `
<div class="input-group input-group--dirty input-group--prepend-icon theme--dark input-group--text-field input-group--single-line input-group--full-width teal--text">
    <div class="input-group__input">
        <i aria-hidden="true" class="icon material-icons input-group__prepend-icon">search</i>
        <input tabindex="0" type="text">
    </div>
    <div class="input-group__details">
        <!---->
    </div>
</div>
`;
